azurerm/resource_arm_api_management_api.go

```diff
@@ -257,16 +260,34 @@ func resourceArmApiManagementApiCreateUpdate(d *schema.ResourceData, meta interf
 		apiType = apimanagement.APIType(apimanagement.SoapToRest)
 	}
 
-	params := apimanagement.APICreateOrUpdateParameter{
-		APICreateOrUpdateProperties: &apimanagement.APICreateOrUpdateProperties{
-			APIType:                       apiType,
-			Description:                   utils.String(description),
-			DisplayName:                   utils.String(displayName),
-			Path:                          utils.String(path),
-			Protocols:                     protocols,
-			ServiceURL:                    utils.String(serviceUrl),
-			SubscriptionKeyParameterNames: subscriptionKeyParameterNames,
-		},
+	var params apimanagement.APICreateOrUpdateParameter
+	if versionSetId != "" {
```

Comment:
rather than doing this, since the only difference between the two blocks appears to be the `versionSetId` field, we could have a single block and do:

```
if versionSetId != "" {
  params.APICreateOrUpdateProperties.APIVersionSetID = utils.String(versionSetId)
}
```

Reply:
Wasn't quite sure how to do this in Go. Thanks for sharing! Changed!